### ISSUE TITLE
is this the desired output for flux-up?

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#855]
Just messing around and noticed

```
$ flux up --newline 
ok:     
[0
1
2
3]
slow:   
fail:   
unknown:
```

Is this what is actually desired?  ```whatsup``` and ```nodeattr``` don't have the brackets b/c presumably the user wants comma/newline output for easy parsing.

As an aside, also noticed no tests for ```flux-up``` either.

**Comment by garlick** [comment]
That's probably a mistake, and some test coverage would be welcome, though `t/t0010-basic-utils.t` does call it.

You may have noticed this command doesn't really do much when the "live" module isn't loaded, which now it isn't by default.  That doesn't mean some tests wouldn't be good to have though.

**Comment by chu11** [comment]
I'll take a crack at this.  Didn't notice it was in the testsuite under generic-utils.  I can add some newline or comma equivalent tests.

**Comment by garlick** [comment]
That would be great!

**Comment by grondo** [comment]
IMO, `--newline` or `--comma` don't make much sense outside of `-u, --up` or `-u, --down`, though I guess it doesn't hurt to allow it. It would also be nice if `--newline` and `--comma` were combined into a `--delim=STRING` argument for more flexibility, but again that is just my opinion.

**Comment by garlick** [comment]
That works for me.  In a bootstrap instance, this could replace `whatsup` so think of it that way?

**Comment by grondo** [comment]
Yes, though eventually it might need a way to map rank to hostname

**Comment by garlick** [comment]
Excellent point.  Too bad hwloc doesn't grab that and drop it in the kvs under `resource.hwloc.by_rank`

**Comment by chu11** [comment]
IIRC, a long time ago the plan was flux could eventually replace cerebro and maybe even other monitoring subsystems.  ```whatsup``` would just "point" to flux to get the equivalent up/down information.

**Comment by chu11** [comment]
ugh, because of the labels (e.g. ```slow: ```) there's some special case handling for newlines if we don't limit ```--delim=STRING``` to just ```--up``` and ```--down```.  I'm inclined to leave ```--comma``` and ```--newline``` as is, as its the style I think we're familiar with.

**Comment by chu11** [comment]
Nevermind.  I originally took the ```--delim=STRING``` literally, as in a random string could be the separator.  But if it's a char, it's no biggie.    We'll go with that.

**Comment by chu11** [comment]
I ended up settling on keeping ```--newline``` and ```--comma``` as the first time I had to do ```--delim=$'\n'```, I thought perhaps this isn't the best option for most end users.

**Comment by chu11** [comment]
The shell won't compile "\n" into a newline character automatically.  ```$''``` is the Bash way to do it make it happen.

**Comment by grondo** [comment]
weird!  I always use it like:
```
 $ hostlist -e -d '\n' foo[1-4]                                                                                                                    
foo1
foo2
foo3
foo4
```

**Comment by chu11** [comment]
Ahh, you're specifically handling it as a special case in ```hostlist```

```
function convert_string_to_escape (s)
        local t = { t='\t', n='\n', f='\f', v='\v', t='\t', r='\r',}
        return s:sub(1,1) == '\\' and t[s:sub(2,2)] or s
end
```

I was about to do something like that, but it felt hackish.  Do you really have your heart set on ```--delim```? :-)

**Comment by grondo** [comment]
No I just had no idea about that limitation! What you are doing is fine with me!

**Comment by chu11** [comment]
Fixed via #858
